### PR TITLE
API changes for EOS-wallets creation: accounts public keys with indexes.

### PR DESCRIPTION
**Pull request by denisshevchenko**
<p align="right">#231</p>

# Overview

- [x] During EOS-wallet creation not only account's public key is provided, but its index as well.

# Comments

Since we receive accounts' public keys as they are, we need their indexes as well.

### CODE REVIEW

File: src/Cardano/Wallet/API/V1/Types.hs [hunk 1]
```diff
@@ -1138,6 +1101,73 @@ instance ToHttpApiData AccountIndex where
...
+        accpubkeywithixPublicKey
+        accpubkeywithixIndex
+
+instance Buildable [AccountPublicKeyWithIx] where
+    build = bprint listJson
+
+-- | A type modelling the request for a new 'EosWallet',
+-- on the mobile client or hardware wallet.
+data NewEosWallet = NewEosWallet
+    { neweoswalAccountsPublicKeysWithIxs :: ![AccountPublicKeyWithIx]
```
Comment (KtorZ) [inline src/Cardano/Wallet/API/V1/Types.hs]: Remember that we are making an API for human to use :upside_down_face:  .. `accountPublicKeys` was fine, or simply, "accounts".

Comment (denisshevchenko) [reply src/Cardano/Wallet/API/V1/Types.hs]: Agree. Fixed.

**Review by KtorZ — APPROVED** [review]

**Comment by KtorZ** [comment]
@denisshevchenko hlint isn't happy with this one ^^^^

**Comment by denisshevchenko** [comment]
@KtorZ Thanks, fixed.

**Review by akegalj — APPROVED** [review]
LGTM

account indexes are not used for anything in EOS wallet so far, am I right? What is a plan on when to use them?

EDIT: I see usage in this PR https://github.com/input-output-hk/cardano-wallet/pull/255/files

**Comment by denisshevchenko** [comment]
> account indexes are not used for anything in EOS wallet so far, am I right?

@akegalj We're storing accounts for EOS-wallets in the same hierarchy, I mean `HdAccount`, and it contains `HdAccountBase` which includes `HdAccountId` which includes `HdAccountIx`. :-) So we _have_ to provide an index.

**Pull request merged**